addons/medical_engine/functions/fnc_handleDamage.sqf

```diff
@@ -97,15 +83,17 @@ if (_hitPoint isEqualTo "ace_hdbracket") exitWith {
     _allDamages sort false;
     (_allDamages select 0) params ["_receivedDamage", "", "_woundedHitPoint"];
 
+    // We know it's structural when no specific hitpoint is damaged
     if (_receivedDamage == 0) then {
         _receivedDamage = _damageStructural;
         _woundedHitPoint = "Body";
     };
     TRACE_2("received",_receivedDamage,_woundedHitPoint);
 
-    // Check for falling damage.
+    // Check for environmental damage sources (EH doesn't explicitly know)
     if (_ammo isEqualTo "") then {
-        if (velocity _unit select 2 < -2) then {
+        // Downward velocity indicates falling damage
+        if ((vehicle _unit == _unit) && {velocity _unit select 2 < -2}) then {
```

Comment:
I think I actually need to swap the fall damage check with the hit by vehicle check (to handle the extreme edge case of being hit by a plane/helicopter while skydiving).

Reply:
This is more a suggestion for code legibility. I like putting brackets to separate commands, etc...: `{((velocity _unit) # 2) < -2 }`

We can  use also `vehicle _unit isEqualTo _unit`

Reply:
The quickest way to check if a unit is on foot is actually `_unit in _unit`, we should probably have added a macro for that to CBA ages ago

Reply:
`OBJECT in OBJECT` is pretty slow.

Reply:
Oh really? I figured because it was one less command it'd be faster 🤷‍♂️